android-build-tools/src/config.rs

```diff
@@ -0,0 +1,60 @@
+use crate::manifest::{Feature, Permission};
+use crate::ndk::Ndk;
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[derive(Debug)]
+pub struct Config {
+    pub ndk: Ndk,
+    pub build_dir: PathBuf,
+    pub package_name: String,
+    pub package_label: String,
+    pub version_name: String,
+    pub version_code: u32,
+    pub target_name: String,
+    pub assets: Option<String>,
+    pub res: Option<String>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct Metadata {
+    pub(crate) target_sdk_version: Option<u32>,
+    pub(crate) min_sdk_version: Option<u32>,
+    pub(crate) icon: Option<String>,
+    pub(crate) fullscreen: Option<bool>,
+    pub(crate) opengles_version: Option<(u8, u8)>,
+    pub(crate) feature: Option<Vec<FeatureConfig>>,
+    pub(crate) permission: Option<Vec<PermissionConfig>>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct FeatureConfig {
+    name: String,
+    required: Option<bool>,
+    version: Option<String>,
```

Comment:
Are what the `version` field means?
I cannot find this in docs.

Reply:
nice catch, I looked at the `cargo-apk` sources when writing the code